extra_data/exceptions.py

```diff
@@ -6,11 +6,13 @@ class FileStructureError(Exception):
 
 
 class SourceNameError(KeyError):
-    def __init__(self, source):
+    # If `custom_message=True`, then `source` is treated as the full error message
+    def __init__(self, source, custom_message=False):
```

Comment:
This falls under nitpicking, but I'd argue `None` is the canonical way to indicate absence of value in Python

Reply:
Hmm, would you then call it as e.g. `SourceNameError(None, "foo")`?

Reply:
Oh, not quite, sorry! I was thinking of:

```python
class SourceNameError(KeyError):
    def __init__(self, source, custom_message=None):
        ...
```

and check for identity with `None` further below.

Concerning how to call it, actually I would be fine to make both arguments optional and use keywords for new invocation. That would keep old code compatible.

Reply:
Alrighty, I think I like the keyword argument style more so I went with that in 2996de4.